server/documents/elements/label.html.eco

```diff
@@ -333,16 +353,115 @@ themes      : ['Default', 'GitHub']
   <div class="example" data-use-content="true" data-class="icon">
     <h4 class="ui header">Icon</h4>
     <p>A label can include an icon</p>
-    <a class="ui label">
+    <div class="ui label">
       <i class="mail icon"></i>
       Mail
-    </a>
-    <a class="ui label">
-      Tag
-      <i class="delete icon"></i>
-    </a>
+    </div>
+    <div class="ui label">
+      <i class="checkmark icon"></i>
+      Test Passed
+    </div>
+    <div class="ui label">
+      <i class="dog icon"></i>
+      Dog
+    </div>
+    <div class="ui label">
+      <i class="cat icon"></i>
+      Cat
+    </div>
+  </div>
+  
+  <div class="another example" data-use-content="true">
+    <div class="ignored ui info message">Icons can be placed in the right side of label.</div>
+
+    <div class="ui right icon label">
+      Mail
+      <i class="mail icon"></i>
+    </div>
+    <div class="ui right icon label">
+      Test Passed
+      <i class="checkmark icon"></i>
+    </div>
+    <div class="ui right icon label">
+      Dog
+      <i class="dog icon"></i>
+    </div>
+    <div class="ui right icon label">
+      Cat
+      <i class="cat icon"></i>
+    </div>
+  </div>
+
+  <div class="another example" data-use-content="true">
+    <div class="ignored ui info message">Icons can be hoverable to indicate an action.</div>
+    <div class="ui label">
+      <i class="hoverable mail icon"></i>
+      Mail
+    </div>
+    <div class="ui label">
+      <i class="hoverable times icon"></i>
+      Test Failed
+    </div>
+
+    <div class="ui right icon label">
+      Mail
+      <i class="hoverable mail icon"></i>
+    </div>
+    <div class="ui right icon label">
+      Test Failed
+      <i class="hoverable times icon"></i>
+    </div>
+  </div>
+
+  <div class="another example">
+    <div class="ignored ui info message">Label can contains icon without text.</div>
+
+    <div class="ui icon label">
+      <i class="mail icon"></i>
+    </div>
+    <div class="ui icon label">
+      <i class="dog icon"></i>
+    </div>
+    <div class="ui icon label">
+      <i class="cat icon"></i>
+    </div>
   </div>
 
+  <div class="another example" data-use-content="true">
+    <div class="ignored ui icon warning message">
+      <i class="warning sign icon"></i>
+      <div class="content">
+        For backward-compatibility during v2.x, <code>close</code>/<code>delete</code> icon...
+      
+        <ul>
+          <li>requires <code>left icon</code> added to parent <code>label</code> when the icon is placed in the left of label.</li>
+          <li>do not require <code>right icon</code> added to parent <code>label</code> when the icon is placed in the right of label.</li>
+          <li>have special style as same as <code>hoverable</code> variation when the icon is inside <code>label</code> which do not have <code>icon</code> class.</li>
+        </ul>
+      
+        This complicated rule may be simplified in v3.x.
```

Comment:
It's more logical than complicated to support a `left icon label` if we imvent a `right icon label`.
Please remove "complicated" and "3.x" to "a later minor or major version"